dev/release-build.sh

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+function exit_with_usage {
+  cat << EOF
+usage: release-build.sh <release-prepare|release-publish|release-snapshot>
+Creates build distributions from a git commit hash.
+
+Top level targets are
+  --release-prepare: Create binary packages and add them to https://dist.apache.org/repos/dist/dev/bahir
+  --release-publish: Publish a release to Apache release repo
+  --release-snapshot: Publish snapshot release to Apache snapshots
+
+  --gitCommitHash - Release tag or commit to build from, default master HEAD
+  --releaseVersion - Release identifier used when publishing
+  --releaseRc - Release RC identifier used when publishing
+  --developmentVersion - Release identifier in top level package directory
+  --tag - Release Tag identifier used when taging git
+
+All other inputs are environment variables
+
+GPG_PASSPHRASE - Passphrase for GPG key
+
+Note: Ensure you have set up standard Apache committer settings in ~/.m2/settings.xml as per
+
+   http://maven.apache.org/developers/committer-settings.html
+
+EOF
+  exit 1
+}
+
+set -e
+
+if [ $# -eq 0 ]; then
+  exit_with_usage
+fi
+
+
+# Process each provided argument configuration
+while [ "${1+defined}" ]; do
+  IFS="=" read -ra PARTS <<< "$1"
+  case "${PARTS[0]}" in
+    --release-prepare)
+      RELEASE_PREPARE=true
+      shift
+      ;;
+    --release-publish)
+      RELEASE_PUBLISH=true
+      shift
+      ;;
+    --release-snapshot)
+      RELEASE_SNAPSHOT=true
+      shift
+      ;;
+    --gitCommitHash)
+      GIT_REF="${PARTS[1]}"
+      shift
+      ;;
+    --releaseVersion)
+      RELEASE_VERSION="${PARTS[1]}"
+      shift
+      ;;
+    --developmentVersion)
+      DEVELOPMENT_VERSION="${PARTS[1]}"
+      shift
+      ;;
+    --releaseRc)
+      RELEASE_RC="${PARTS[1]}"
+      shift
+      ;;
+    --tag)
+      RELEASE_TAG="${PARTS[1]}"
+      shift
+      ;;
+    *help* | -h)
+      exit_with_usage
+     exit 0
+     ;;
+    -*)
+     echo "Error: Unknown option: $1" >&2
+     exit 1
+     ;;
+    *)  # No more options
+     break
+     ;;
+  esac
+done
+
+
+for env in GPG_PASSPHRASE; do
```

Comment:
Will leave like this, as it's ready to support multiple env variables if needed in the future